File: cases/taint/cpp/local-chain-positive/local_chain_positive.cpp
int dfb_source() { // DFB-SOURCE: local-chain-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: local-chain-sink

void run() {
    int first = dfb_source();
    int second = first; // DFB-WITNESS: local-chain-second
    int third = second; // DFB-WITNESS: local-chain-third
    dfb_sink(third);
}

File: cases/taint/scala/local-chain-positive/LocalChainPositive.scala
package dataflowbench

object LocalChainPositive {
  def dfb_source(): Int = { // DFB-SOURCE: local-chain-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: local-chain-sink

  def run(): Unit = {
    val first = dfb_source()
    val second = first // DFB-WITNESS: local-chain-second
    val third = second // DFB-WITNESS: local-chain-third
    dfb_sink(third)
  }
}
